scripts/shared/cloud-prepare.sh

```diff
@@ -28,15 +28,21 @@ function check_gateway_exists() {
 }
 
 function prepare_kind() {
-    read -r -a nodes <<< "${cluster_nodes[$cluster]}"
-    local node=${cluster}-${nodes[-1]}
-    kubectl label node "$node" "$GATEWAY_LABEL" --overwrite
-
-    if [[ "$AIR_GAPPED" = true ]]; then
-        local pub_ip
-        pub_ip=$(kubectl get nodes "$node" -o jsonpath="{.status.addresses[0].address}")
-        kubectl annotate node "$node" gateway.submariner.io/public-ip=ipv4:"$pub_ip"
-    fi
+    local gw_count
+    local gw_nodes
+    gw_count="${gateways[$cluster]:-1}"
+    gw_nodes=$(kubectl get nodes --no-headers=true \
+        -o custom-columns=NAME:.metadata.name | tail -n "$gw_count")
+
+    for gw_node in ${gw_nodes}; do
+        kubectl label node "$gw_node" "$GATEWAY_LABEL" --overwrite
```

Comment:
```suggestion
    for node in "${nodes[@]:0:$gw_count}"; do
        kubectl label node "$node" "$GATEWAY_LABEL" --overwrite
```

Reply:
@mkolesnik 
I tried to apply this suggestion, but apparently it look like this way of looping, with control of the range in bash is working only for integers and not strings.